bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -0,0 +1,290 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecast;
+import org.openhab.binding.solarforecast.internal.utils.Utils;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ForecastSolarObject} holds complete data for forecast
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ForecastSolarObject implements SolarForecast {
+    private static final double UNDEF = -1;
+
+    private final Logger logger = LoggerFactory.getLogger(ForecastSolarObject.class);
+    private final TreeMap<ZonedDateTime, Double> wattHourMap = new TreeMap<ZonedDateTime, Double>();
+    private final TreeMap<ZonedDateTime, Double> wattMap = new TreeMap<ZonedDateTime, Double>();
+    private final DateTimeFormatter dateInputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private ZoneId zone = ZoneId.systemDefault();
+    private Optional<String> rawData = Optional.empty();
+    private boolean valid = false;
+    private Instant expirationDateTime;
+
+    public ForecastSolarObject() {
+        expirationDateTime = Instant.now();
+    }
+
+    public ForecastSolarObject(String content, Instant expirationDate) {
+        expirationDateTime = expirationDate;
+        if (!content.isEmpty()) {
+            rawData = Optional.of(content);
+            try {
+                JSONObject contentJson = new JSONObject(content);
+                JSONObject resultJson = contentJson.getJSONObject("result");
+                JSONObject wattHourJson = resultJson.getJSONObject("watt_hours");
+                JSONObject wattJson = resultJson.getJSONObject("watts");
+                String zoneStr = contentJson.getJSONObject("message").getJSONObject("info").getString("timezone");
+                zone = ZoneId.of(zoneStr);
+                Iterator<String> iter = wattHourJson.keys();
+                // put all values of the current day into sorted tree map
+                while (iter.hasNext()) {
+                    String dateStr = iter.next();
+                    // convert date time into machine readable format
+                    try {
+                        ZonedDateTime zdt = LocalDateTime.parse(dateStr, dateInputFormatter).atZone(zone);
+                        wattHourMap.put(zdt, wattHourJson.getDouble(dateStr));
+                        wattMap.put(zdt, wattJson.getDouble(dateStr));
+                    } catch (DateTimeParseException dtpe) {
+                        logger.warn("Exception parsing time {} Reason: {}", dateStr, dtpe.getMessage());
```

Comment:
Does it still make sense to create the object in this case?

Reply:
Code belongs to constructor so Object is already created. In case of error during content parsing the `valid `flag is false and will trigger a new forecast fetch.
Btw: never saw this error during testing

Reply:
Done